sidechain/consensus/common/src/block_import.rs

```diff
@@ -118,21 +118,14 @@ where
 			signed_sidechain_block.block().block_data().layer_one_head()
 		);
 
-		let peeked_parentchain_header =
-			self.peek_parentchain_header(&sidechain_block, parentchain_header)
-				.unwrap_or_else(|e| {
-					warn!("Could not peek parentchain block, returning latest parentchain block ({:?})", e);
-					parentchain_header.clone()
-				});
+		// Peeking the parentchain header to ensure the sidechain block points to a valid parentchain block.
+		let _peeked_parentchain_header =
```

Comment:
the pc header peeking returns an error in case the pc header does not match the sidechain header. If that's the case, we simply return here, and the `ensure!` in `verifier.verify` can be dropped completely without leaving an opening.